source-git-monorepos/README.md

```diff
@@ -50,21 +89,367 @@ packages:
 
 With the above source-git configuration could be kept in
 `.distro/source-git.yaml`. Downstream files to be synced to dist-git are to be
-kept in subdirectories, named according to the distro package, in `.distro` to
-keep things organized.
+kept in subdirectories, named according to `package_name` (the key used in
+`source-git.yaml` to reference the package), in `.distro`, to keep things
+organized.
+
+An example layout:
+
+```
+.distro
+├── backend
+│   └── copr-backend.spec
+├── cli
+│   └── copr-cli.spec
+├── frontend
+│   └── copr-frontend.spec
+└── source-git.yaml
+```
 
 This way all distribution configs and files would be kept at one place, which
-probably is somewhat better than having source-git configuration spread
-across the monorepo, in subdirectories. This setup also allows multiple paths
-from the repo mapped to a dist-git repo.
+probably is somewhat better than having source-git configuration spread across
+the monorepo, in subdirectories. This setup also allows multiple paths from
+the repo mapped to a dist-git repo.
+
+### `packit source-git init`
+
+If the source-git repository to be initialized is a monorepo, then each
+dist-git repository specified will also need to one or more subdirectories of
+the source-git repository.
+
+Which makes the whole thing a little bit too complicated ☺
+
+This complication could be elevated, if `packit source-git init` would use a
+guided (dialog-like) process, instead of taking all its inputs as arguments.
+
+As an alternative, the command could have a variant where it takes a basic
+`packages` definition to guide the initialization of a source-git monorepo.
+
+Example `packages.yaml`:
+
+```yaml
+- downstream_package_name: copr-backend
+  package_path: backend
+  dist_git_path: rpms/copr-backend
+  dist_git_branch: rawhide
```

Comment:
I tend to feel that this should be a list of branch names?

Reply:
Why so?

`source-git init` initializes a single branch in a source-git repo. Furthermore: source-git branches map to dist-git branches one-to-one.

Reply:
Hm, I guess? I'm offtopic (I thought it is the shared config format with the packit service, and the "propose downstream" configuration).

Just from the experience with tito... where you configure the precise set of branches you want to automatically update with new releases (aka `tito release`).

Reply:
This configuration is a proposal that could make **initializing** source-git monorepos from dist-git repositories easier. It's not planned to be used as a source-git config per se.